universal-application-tool-0.0.1/app/controllers/dev/AzuriteFileStorageStrategy.java

```diff
@@ -16,19 +16,27 @@ public Result create(StoredFileRepository storedFileRepository, Request request)
     Optional<String> etag = request.queryString("etag");
     Optional<String> container = request.queryString("container");
     Optional<String> fileName = request.queryString("fileName");
-    Optional<String> userFileName = request.queryString("userFileName");
-    updateFileRecord(storedFileRepository, fileName.get());
+    Optional<String> originalFileName = request.queryString("originalFileName");
+    updateFileRecord(storedFileRepository, fileName.get(), originalFileName.get());
     String successMessage =
         String.format(
             "File successfully uploaded to Azure: container: %s, file name: %s, etag: %s, user"
                 + " file name: %s.",
-            container.get(), fileName.get(), etag.orElse(""), userFileName.get());
+            container.get(), fileName.get(), etag.orElse(""), originalFileName.get());
     return redirect(routes.FileUploadController.index().url()).flashing("success", successMessage);
   }
 
-  private void updateFileRecord(StoredFileRepository storedFileRepository, String key) {
+  private void updateFileRecord(
+      StoredFileRepository storedFileRepository, String key, String originalFileName) {
     StoredFile storedFile = new StoredFile();
     storedFile.setName(key);
+    storedFile.setOriginalFileName(getPrefixedUserFileName(key, originalFileName));
     storedFileRepository.insert(storedFile);
   }
+
+  private String getPrefixedUserFileName(String fileName, String originalFileName) {
```

Comment:
suggestion (blocking): I'm seeing the concept of `PrefixedUserFileName` referenced in a few other parts of this PR and it's not clear to me what the concept is. Please document it in comments.

question (blocking): Is it a concept that will only ever be in dev or will it be used in production code as well?

Reply:
Added a comment. And yes, this will be used in production code too. The prefixedOriginalFileName will either be in the format "dev/${filename}" or "applicant-%d/program-%d/block-%s/${filename}" where $filename is the original name of the file. depending on whether the file is being uploaded from an application or the dev upload view.